### DISCUSSION TITLE
2.8.9 更新日志

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#4959] [讨论]
### 更新亮点
- 支持下载原版服务端（#112，@WTP016-CN，PR @tangge233）
    <img width="480" src="https://github.com/user-attachments/assets/b4d9e2ed-1d65-4928-b31d-971f086252bf">

- 本地 Mod 的标题支持选择显示 Mod 原始文件名（#3073，@xingchen-Forever）
    <img width="509" src="https://github.com/user-attachments/assets/20217a32-91ba-4037-88f4-de53e4ca59b9">

- 新主页预设：PCL 新功能说明书（#4905，由 @WForst-Breeze 制作）
    <img width="566" src="https://github.com/user-attachments/assets/888768d2-69de-4097-a55e-49566371b2be">

- 新主页预设：OpenMCIM 仪表盘（#4906，由 @SALTWOOD 制作）
    <img width="559" src="https://github.com/user-attachments/assets/395c68c8-2a46-41f9-9dc4-39855bba7f82">

### 其他更新内容
- 重命名时禁止输入 NTFS 8.3 文件名（#4505，@AptS-1547，PR @SALTWOOD）
- 搜索筛选 1.13- 且为 Forge 的 Mod 时，视作没有设置 Mod 加载器的筛选项，以应对部分老 Mod 作者没有设置 Mod 兼容的加载器的情况（#4512，@MinecraftYYDS7）
- 将启动器窗口大小存储在文件而非注册表，以便分发整合包（#4678，@LuckiQ）
- 安装 Forge/NeoForge 时会自动新建 mods 文件夹（#4898，@ThendJyc）
- 为指向百度的下载添加了独特的 UA（#4951，@kazukokawagawa）
- 添加 [使用密码登录正版账号] 帮助（LTCatt/PCL2Help#395，PR @tangge233）
- 将 Mod 与整合包的设置独立归类进一张卡片中
- 减淡了本地 Mod 项的标签和描述的颜色，以便与标题区分
- 移除安装整合包时保留原始压缩包选项
- 支持分析误删 Forge 安装后的文件导致的崩溃
### 技术性更改
- 添加 MyImage 控件（#4868，PR @tangge233）
    该控件基于 WPF 原生的 Image，支持 WebP 格式，支持缓存
    MyListItem、MyCompItem、MyLocalModItem 的图标换用 MyImage 控件
    如果你编写了自定义主页或帮助，建议将所有 Image 控件替换为 local:MyImage，并将图片格式更换为 WebP，这可以大大节省流量
    你可以通过网络请求的 Header 判断 PCL 的版本，以实现仅在新版本启用 MyImage 控件
    新增的属性详见：https://github.com/Hex-Dragon/PCL2/blob/main/Plain%20Craft%20Launcher%202/Resources/Custom.xaml#L329
- EventType [下载文件] 支持指定文件名和目标文件夹（#4737，@wuliaodexiaoluo）
- 若加载的图片不是 WebP 格式，则不加载 WebP DLL，以优化性能
- PCL 标题栏、MC 皮肤支持选择 WebP 格式的图片
### Bug 修复
- 对 Mod 进行操作后，Mod 管理列表会不必要地刷新（#4457，@sinterde）
- 罕见情况下，离开 Mod 管理页面导致出错（#4721，@YIEMM）
- 用户名有特殊字符时无法使用 LittleSkin 启动游戏（#4777，@Crotina）
- 1.0.0-rc2-2 在下载页面错误地被归类到了快照版（#4803，@Crotina）
- 拖拽加载主页后设置单选框未刷新（#4826，@JingHai-Lingyun）
- 关闭动画时无法解锁某个隐藏主题（#4830，@Joker2184）
- 打开老版本 PCL 后需要重新进行登录（#4836，@Apple-alone）
- 搜索后启用/禁用 Mod 时出错（#4862，@EasyGuo114514）
- 中文路径、中文用户名下 PCL 可能出错（#4869，@h7f8）
- 对一个项的两个不同信息源进行搜索时可能遗漏搜索结果（#4912，@ChenXi094）
- 热力系列 Mod 中文名显示错误（#4916，@ChenMoFeiJin）
- 部分崩溃会给出奇怪的分析结果（#4936，@lactobionicAcid）
- 极端情况下，在下载刚结束时删除版本导致出错（#4949，@lunardream2）
- 同时加载多个相同图片可能出错
### 下一个版本的更新计划
- 尽量完成除导出整合包以外的 中优先度 提交

<hr /><em>This discussion was created from the release <a href='https://github.com/Hex-Dragon/PCL2/releases/tag/2.8.9'>2.8.9</a>.</em>

**Comment by Joker2184** [comment]
PCL2 更新日志 已同步至2.8.9.jpg
你可以通过预设获取最新的PCL2新闻

**Reply by llsgllsg** [reply]
![image](https://github.com/user-attachments/assets/39d04569-93d3-41c9-8c41-bf7751828528)
福建的痛，很多预设根本没法访问啊啊啊啊啊

**Reply by tangge233** [reply]
> 福建的痛，很多预设根本没法访问啊啊啊啊啊

请坐和放宽，社区正在努力换源

**Reply by Joker2184** [reply]
> ![image](https://private-user-images.githubusercontent.com/151439431/380492755-39d04569-93d3-41c9-8c41-bf7751828528.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzIwMDgsIm5iZiI6MTczMDAzMTcwOCwicGF0aCI6Ii8xNTE0Mzk0MzEvMzgwNDkyNzU1LTM5ZDA0NTY5LTkzZDMtNDFjOS04YzQxLWJmNzc1MTgyODUyOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QxMjIxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjVhM2EzN2Q2NDk3OThiNDU3Zjc2ZDQ5MTEzZGI5ZjQ2ZGRmODhhYjU4MzJmMDEyYzI5MzhmYTY5N2I3YmE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7kF_s4bUv_ScpldbsOqNLbbJen2d1ES8ky1n9LuS454) 福建的痛，很多预设根本没法访问啊啊啊啊啊

使用CF的作者们已经在讨论换源
你可以通过https://github.com/Hex-Dragon/PCL2/discussions/5014查看进度
请稍等~

**Reply by llsgllsg** [reply]
我觉得挂在雨云的免费对象存储是个正确的事
![image](https://github.com/user-attachments/assets/f8ee69a7-6a24-43cd-ac81-dce44bf666cb)

**Reply by Joker2184** [reply]
> 我觉得挂在雨云的免费对象存储是个正确的事 ![image](https://private-user-images.githubusercontent.com/151439431/380497255-f8ee69a7-6a24-43cd-ac81-dce44bf666cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzMyNzksIm5iZiI6MTczMDAzMjk3OSwicGF0aCI6Ii8xNTE0Mzk0MzEvMzgwNDk3MjU1LWY4ZWU2OWE3LTZhMjQtNDNjZC1hYzgxLWRjZTQ0YmY2NjZjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QxMjQyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMDM4NmZhOTk3OGU2MDdjYjc1ZmJmMzFmMzQwZDJmMDQ5MmM3MTdiODc4M2QxMmRiYTVmMDU2YmFhZjljMDc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EXMzXqutx_hvqupqcccRTNWpSkzv9p-UXK6p51BPmac)

可能吧？但是集资可以考虑年付或者服务器了

在线构建也可以考虑了 流量不限（

**Comment by SqrtTan** [comment]
> 下一个版本的更新计划
> - 尽量完成除导出整合包以外的 中优先度 提交

立Flag了.jpg

**Comment by Orangesmart** [comment]
启动器好像没有反应过来有新版本
![image](https://github.com/user-attachments/assets/ac5022d5-df91-4a59-9cd3-2ac0afd3532b)

**Reply by Joker2184** [reply]
本次更新是快照版
Orz
